src/ui/components/checkbox-group.ts

```diff
@@ -0,0 +1,79 @@
+import { css, html, LitElement, type TemplateResult } from "lit";
+import { customElement } from "lit/decorators.js";
+import { repeat } from "lit/directives/repeat.js";
+
+import { Input } from "../mixins/input";
+import { List } from "../mixins/list";
+import { SDCheckboxElement } from "./checkbox";
+import { type SDOptionElement } from "./option";
+
+/**
+ * Element that offers persisting an set of values, from a group of checkbox options.
+ */
+@customElement("sd-checkboxgroup")
+export class SDCheckboxGroupElement extends List(Input<(boolean | number | string)[]>(LitElement)) {
+	/**
+	 * @inheritdoc
+	 */
+	public static styles = [
+		super.styles ?? [],
+		css`
+			sd-checkbox {
+				display: flex;
+			}
+		`,
+	];
+
+	/**
+	 * @inheritdoc
+	 */
+	public override render(): TemplateResult {
+		return html`
```

Comment:
It looks like in a `<sd-radiogroup>` the label or the slot content from the inner `<sd-option>` is displayed, but in a `<sd-checkboxgroup>` only slot content from the inner `<sd-option>` is displayed. I think we should also be looking for the `label` property of the `<sd-option>` here?

```html
<sd-field label="Radio Group">
	<sd-radiogroup setting="radioSetting">
		<sd-option label="Value 1" value="1">Other Value</sd-option>
		<sd-option label="Value 2" value="2"></sd-option>
		<sd-option label="Value 3" value="3"></sd-option>
	</sd-radiogroup>
</sd-field>
<sd-field label="Checkbox Group">
	<sd-checkboxgroup setting="checkboxGroupSetting">
		<sd-option label="Value 1" value="1">Other Value</sd-option>
		<sd-option label="Value 2" value="2"></sd-option>
		<sd-option label="Value 3" value="3"></sd-option>
	</sd-checkboxgroup>
</sd-field>
```
![CleanShot 2024-11-06 at 16 02 09@2x](https://github.com/user-attachments/assets/27578e5e-3140-4e0a-9894-0fb9ece410f3)

Reply:
Great call out; I've updated the following option-type elements to utilize their `textContent` as the `label`:
- `<sd-checkbox />`
- `<sd-option />`
- `<sd-radio />`
- `<sd-switch />`
This allows for the elements to be constructed in a similar manor to `<select />`, for example
```html
<sd-switch setting="isOn">On / Off</sd-switch>
```
Or in the context of a group:
```html
<sd-checkboxgroup setting="options">
    <sd-option value="1">One</sd-option>
    <sd-option value="2">Two</sd-option>
</sd-checkboxgroup>
```